picard/log.py

```diff
@@ -55,20 +55,22 @@ def debug_mode(enabled):
 
 class TailLogHandler(logging.Handler):
 
-    def __init__(self, log_queue, tail_logger):
+    def __init__(self, log_queue, tail_logger, log_queue_lock):
         super().__init__()
         self.log_queue = log_queue
         self.tail_logger = tail_logger
+        self.log_queue_lock = log_queue_lock
         self.pos = 0
 
     def emit(self, record):
-        self.log_queue.append(
-            TailLogTuple(
-                self.pos,
-                self.format(record),
-                record.levelno
+        with self.log_queue_lock:
+            self.log_queue.append(
+                TailLogTuple(
+                    self.pos,
+                    self.format(record),
+                    record.levelno
+                )
             )
-        )
         self.pos += 1
```

Comment:
Does this line need to be protected by a lock as well, since `+=` is not atomic in Python, AFAIR.